### PR TITLE
Add mattermost adapter (#49)

### PR DESCRIPTION
**Pull request by bergo**
Created a mattermost adpater, based on the telegram adapter with a table of listing results

### CODE REVIEW

File: lib/notification/adapter/mattermost.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+/**
+ * sends new listings to mattermost
+ * @param serviceName e.g immowelt
+ * @param newListings an array with newly found listings
+ * @param notificationConfig config of this notification adapter
+ * @param jobKey name of the current job that is being executed
+ * @returns {Promise<Void> | void}
+ */
+exports.send = ({ serviceName, newListings, notificationConfig, jobKey }) => {
```
Comment (orangecoding) [inline lib/notification/adapter/mattermost.js:13]: As far as I understood, this adapter sends messages per chat? If so, it is useful to use the telegram adapter, which shortens the results? Mattermost should not have the restriction of 30 chars no?

File: lib/notification/adapter/mattermost.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  readme: markdown2Html('lib/notification/adapter/mattermost.md'),
+  description: 'Fredy will send new listings to your mattermost team chat.',
+  fields: {
+    webhook: {
+      type: 'text',
+      label: 'Webhook',
+      description: 'The incoming webhook url',
+    },
+    channel: {
+      type: 'channel',
```
Comment (orangecoding) [inline lib/notification/adapter/mattermost.js:54]: the type is allways text (or number, but in your case it's text)

**Review by orangecoding — CHANGES_REQUESTED** [review]

**Comment by orangecoding** [comment]
thanks for this pr, please check the 2 comments.

File: lib/notification/adapter/mattermost.js (hunk 1 above)
Comment (bergo) [reply lib/notification/adapter/mattermost.js:13]: Yes, thats true. I used it for better mobile view, but works without in the same way.  Changed it.

File: lib/notification/adapter/mattermost.js (hunk 2 above)
Comment (bergo) [reply lib/notification/adapter/mattermost.js]: Ok, changed to text-type

**Pull request merged**